src/main/java/com/github/jparkie/pdd/BitArray.java

```diff
@@ -39,21 +39,27 @@ private static int numWords(long numBits) {
     }
 
     public boolean get(long index) {
-        return (data[(int) (index >>> 6)] & (1L << index)) != 0;
+        final int arrayIndex = (int)(index >>> 6);
+        final long bitMask = 1L << (index & 0x3F);
+        return (data[arrayIndex] & bitMask) != 0;
     }
 
     public boolean set(long index) {
-        if (!get(index)) {
-            data[(int) (index >>> 6)] |= (1L << index);
+        final int arrayIndex = (int)(index >>> 6);
+        final long bitMask = 1L << (index & 0x3F);
+        if ((data[arrayIndex] & bitMask) == 0) {
+            data[arrayIndex] |= bitMask;
             bitCount++;
             return true;
         }
         return false;
     }
 
     public boolean clear(long index) {
-        if (get(index)) {
-            data[(int) (index >>> 6)] &= ~(1L << index);
+        final int arrayIndex = (int)(index >>> 6);
+        final long bitMask = 1L << (index & 0x3F);
```

Comment:
You don't need to mask with `0x3F`. According to http://docs.oracle.com/javase/specs/jls/se7/html/jls-15.html#jls-15.19, it is done automatically.